test-network-function/diagnostic/suite.go

```diff
@@ -355,3 +367,29 @@ func listClusterCSIInfo() {
 	err = json.Unmarshal([]byte(match.Match), &csiDriver)
 	gomega.Expect(err).To(gomega.BeNil())
 }
+
+func testCrds(env *config.TestEnvironment) {
+	var labels []string
+
+	for _, configLabel := range env.Config.TargetPodLabels {
+		labels = append(labels, autodiscover.BuildLabelQuery(configLabel))
+	}
+
+	handler := crdstatus.NewCrdStatus(defaultTestTimeout, labels)
+	test, err := tnf.NewTest(common.GetContext().GetExpecter(), handler, []reel.Handler{handler}, common.GetContext().GetErrorChannel())
+	gomega.Expect(err).To(gomega.BeNil())
+	testResult, err := test.Run()
+	gomega.Expect(testResult).To(gomega.Equal(tnf.SUCCESS))
+	gomega.Expect(err).To(gomega.BeNil())
+
+	if len(handler.CrdItems) == 0 {
+		ginkgo.Skip("No CRDs found using target labels.")
+	}
+
+	for _, crd := range handler.CrdItems {
+		ginkgo.By("Checking CRD: " + crd.Name)
+		if crd.Status == nil {
+			ginkgo.Fail("CRD " + crd.Name + " does not have status spec.")
+		}
+	}
+}
```

Comment:
few questions here :
do we want this to be in the diagnostic suite ? if yes, do we want to add any of this information to the claim file ?

Reply:
Salah is right. This shouldn't be in diagnostic. It was the "dumping all CRD" alternative that should be part of diagostic. I guess this one could fall into "obsevability"?